android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/activity/GeckoViewActivityContextDelegate.kt

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.activity
+
+import android.app.Activity
+import android.content.Context
+import mozilla.components.support.base.log.Log
+import org.mozilla.geckoview.GeckoView
+import java.lang.ref.WeakReference
+
+private const val LOGTAG = "GeckoViewActivityContextDelegate"
+
+/**
+ * GeckoViewActivityContextDelegate provides an active Activity to GeckoView or null. Not to be confused
+ * with the runtime delegate of [GeckoActivityDelegate], which is tightly coupled to webauthn.
+ * See bug 1806191 for more information on delegate differences.
+ *
+ * @param contextRef A reference to an active Activity context or null for GeckoView to use.
+ */
+class GeckoViewActivityContextDelegate
+(private val contextRef: WeakReference<Context?>) : GeckoView.ActivityContextDelegate {
+    init {
+        if (contextRef.get() == null) {
+            Log.log(
```

Comment:
We should use the AC (support-base) Logger component here. You can create an instance as `Logger("GeckoViewActivityContextDelegate")` and use that around.

For context, this helps because we can output to different places besides the Android logcat.

There are some examples [here](https://searchfox.org/mozilla-mobile/search?q=Logger&path=firefox-android%2F**%2F*.kt&case=false&regexp=false).

Reply:
Thanks!